runtime/connectors/connectors.go

```diff
@@ -77,8 +77,13 @@ func (ps PropertySchema) ValidateType(val any) bool {
 // Env contains contextual information for a source, such as the repo it came from
 // and (in the future) secrets configured by the user.
 type Env struct {
-	RepoDriver string
-	RepoDSN    string
+	RepoDriver         string
+	RepoDSN            string
+	UseHostCredentials bool // flag to disable use of host credentials
+	AccessKeyID        string
+	SecretAccessKey    string
+	SessionToken       string
+	SecretGlob         []byte
```

Comment:
Since `Env` is generic for all connectors, it should just pass the KV-pairs here, e.g.: `Variables map[string]string`. Then parsing into fields can happen in the actual connector (depending on what fields it expects)